src/InboxNotifications.php

```diff
@@ -0,0 +1,134 @@
+		// 'woocommerce-gutenberg-products-block', // Disallow the notification if the store is using the feature plugin already.
+		'woocommerce-all-products-for-subscriptions',
+		'woocommerce-bookings',
+		'woocommerce-box-office',
+		'woocommerce-cart-add-ons',
+		'woocommerce-checkout-add-ons',
+		'woocommerce-checkout-field-editor',
+		'woocommerce-conditional-shipping-and-payments',
+		'woocommerce-dynamic-pricing',
+		'woocommerce-eu-vat-number',
+		'woocommerce-follow-up-emails',
+		'woocommerce-gateway-amazon-payments-advanced',
+		'woocommerce-gateway-authorize-net-cim',
+		'woocommerce-google-analytics-pro',
+		'woocommerce-memberships',
+		'woocommerce-paypal-payments',
+		'woocommerce-pre-orders',
+		'woocommerce-product-bundles',
+		'woocommerce-shipping-fedex',
+		'woocommerce-smart-coupons',
+	];
+	const ELIGIBLE_COUNTRIES                       = [
+		'GB',
+		'US',
+	];
+
+	/**
+	 * Creates a notification letting merchants know about the Cart and Checkout Blocks.
+	 */
+	public static function create_surface_cart_checkout_blocks_notification() {
+		// If this is the feature plugin, then we don't need to do this. This should only show when Blocks is bundled
+		// with WooCommerce Core.
+		if ( false && Package::feature()->is_feature_plugin_build() ) {
+			return;
+		}
+
+		// Pick a random number between 1 and 100 and add this to the wp_options table. This can then be used to target
+		// a percentage of users.
+		$existing_probability = get_option( self::SURFACE_CART_CHECKOUT_PROBABILITY_OPTION );
+		if ( false === $existing_probability ) {
+			$existing_probability = wp_rand( 0, 100 );
+			add_option( self::SURFACE_CART_CHECKOUT_PROBABILITY_OPTION, $existing_probability );
+		}
+		if ( ! class_exists( 'Automattic\WooCommerce\Admin\Notes\WC_Admin_Notes' ) ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WC_Data_Store' ) ) {
+			return;
+		}
+
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+		$note_ids   = $data_store->get_notes_with_name( self::SURFACE_CART_CHECKOUT_NOTE_NAME );
+
+		foreach ( (array) $note_ids as $note_id ) {
+			$note = Notes::get_note( $note_id );
+
+			// Return now because the note already exists.
+			if ( $note->get_name() === self::SURFACE_CART_CHECKOUT_NOTE_NAME ) {
+				return;
+			}
+		}
+
+		// Calculate store's eligibility to be shown the notice, starting with whether they have any plugins we know to
+		// be incompatible with Blocks.
+		foreach ( self::INELIGIBLE_EXTENSIONS as $extension ) {
+			if ( is_plugin_active( $extension . '/' . $extension . '.php' ) ) {
+				return;
+			}
+		}
+
+		// Next check the store is located in one of the eligible countries.
+		$raw_country = get_option( 'woocommerce_default_country' );
+		$country     = explode( ':', $raw_country )[0];
+		if ( ! in_array( $country, self::ELIGIBLE_COUNTRIES, true ) ) {
+			return;
+		}
+
+		// Finally, check if the store's generated % chance is below the % of users we want to surface this to.
```

Comment:
Would it make sense to check this earlier? If we're excluding 90% of users to start with, we might not want to run the previous checks for that 90% of users.

Maybe we could consider changing the order of some of the checks i.e. the ones that exclude the most users first. For example, given the current set of countries, at first glance, it seems like that check would limit the set of eligible users more than the limited list of incompatible extensions.